internal/pkg/service/common/configmap/configmap_test.go

```diff
@@ -0,0 +1,100 @@
+package configmap
+
+import (
+	"net/netip"
+	"net/url"
+	"time"
+)
+
+type CustomStringType string
+
+type CustomIntType int
+
+type TestConfigWithValidationError struct {
+	Foo             string `configKey:"foo"`
+	ValidationError error
+}
+
+func (c TestConfigWithValidationError) Normalize() {
+	// nop
+}
+
+func (c TestConfigWithValidationError) Validate() error {
+	return c.ValidationError
+}
+
+type TestConfig struct {
+	Embedded         `configKey:",squash"`
+	Ignored          string
+	CustomString     CustomStringType `configKey:"customString"`
+	CustomInt        CustomIntType    `configKey:"customInt"`
+	SensitiveString  string           `configKey:"sensitiveString" sensitive:"true"`
+	StringSlice      []string         `configKey:"stringSlice"`
+	Int              int              `configKey:"int"`
+	IntSlice         []int            `configKey:"intSlice"`
+	Float            float64          `configKey:"float"`
+	StringWithUsage  string           `configKey:"stringWithUsage" configUsage:"An usage text."`
+	Duration         time.Duration    `configKey:"duration"`
+	DurationNullable *time.Duration   `configKey:"durationNullable"`
+	URL              *url.URL         `configKey:"url"`
+	Addr             netip.Addr       `configKey:"address"`         // TextUnmarshaler/BinaryUnmarshaler interface
+	AddrNullable     *netip.Addr      `configKey:"addressNullable"` // TextUnmarshaler/BinaryUnmarshaler interface
+	Nested           Nested           `configKey:"nested"`
+}
+
+func (c TestConfig) Normalize() {
+	// nop
+}
+
+func (c TestConfig) Validate() error {
+	// nop
+	return nil
+}
+
+type TestConfigWithValueStruct struct {
+	EmbeddedValue    `configKey:",squash"`
+	Ignored          Value[string]
+	CustomString     Value[CustomStringType] `configKey:"customString"`
+	CustomInt        Value[CustomIntType]    `configKey:"customInt"`
```

Comment:
Ak sa pouzije genericky typ `Value[T]`, tak je mozne zistit `Value.SetBy`, teda ako bola hodnota nastavena (cez flag, ENV, ...).

Reply:
Pouzivame to napr. v CLI, kde je interaktivny dialog.

Niektore otazky v dialogu preskakujeme, ked zistime, ze si hodnotu uz nastavil cez flag.

Kedze hodnota moze mat default value (ktora sa pouzije v ne-interaktivnom rezime), tak nestaci iba skontrolovat, ci nie je prazdna.